extensions/amp-subscriptions/0.1/local-subscription-platform-iframe.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {Entitlement} from './entitlement';
+import {LocalSubscriptionBasePlatform}
+  from './local-subscription-platform-base';
+import {Messenger} from '../../amp-access/0.1/iframe-api/messenger';
+import {assertHttpsUrl, parseUrlDeprecated} from '../../../src/url';
+import {devAssert, userAssert} from '../../../src/log';
+import {isArray} from '../../../src/types';
+import {parseJson} from '../../../src/json';
+import {toggle} from '../../../src/style';
+
+
+/**
+ * Implments the iframe local subscriptions platform which provides
+ * authorization and pingback via an iframe
+ *
+ * @implements {./subscription-platform.SubscriptionPlatform}
```

Comment:
Implied from the class extending base. But fine to keep if you want to make it obvious.